packages/modeling/src/operations/booleans/martinez/connect_edges.js

```diff
@@ -99,7 +99,10 @@ const initializeContourFromContext = (event, contours, contourId) => {
     // result".
     const lowerContourId = prevInResult.outputContourId
     const lowerResultTransition = prevInResult.resultTransition
-    if (lowerResultTransition > 0) {
+    if (lowerContourId < 0) {
```

Comment:
@platypii I hate to ask this but I really feel that there will be other fixes to the Martinez code base. Therefore, let's start this off right with some a test case for this fix. I saw the new issue at the Martinez repo so hopefully this will be easy.

Reply:
Already done!

One new test in `project.test.js` that demonstrates the bug using `torus({ innerSegments: 8, outerSegments: 4})`

The other is in `unionGeom2.test.js` which is a simplified example extracted from the torus test. This is the same one that I submitted to the `martinez` repo.

I intend to improve both JSCAD and Martinez as a result of this integration :smiley: